app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java

```diff
@@ -207,27 +151,21 @@ public String getDefaultUserAgent(int mode) {
 
     @Override
     public void exitFullScreen() {
-        if (mFullScreenCallback != null) {
-            mFullScreenCallback.exitFullscreen();
-        }
+        // TODO: implement
     }
 
     @NonNull
     @Override
     public WDisplay acquireDisplay() {
-        assert mDisplay == null;
-        assert mTab != null;
-        mDisplay = new DisplayImpl(mRuntime.acquireDisplay(mSettings.getUsePrivateMode()), this);
-        mDisplay.getBrowserDisplay().getBrowser().addTab(mTab);
-        mDisplay.getBrowserDisplay().getBrowser().setActiveTab(mTab);
+        WebContents webContents = mRuntime.createWebContents();
+        BrowserDisplay display = createBrowserDisplay(webContents);
+        mWindowAndroid.setAnimationPlaceholderView(mContentViewRenderView);
+        mDisplay = new DisplayImpl(display, this, mContentViewRenderView);
         return mDisplay;
     }
 
     @Override
     public void releaseDisplay(@NonNull WDisplay display) {
-        assert mDisplay != null;
-        mRuntime.releaseDisplay(mDisplay.getBrowserDisplay());
-        mDisplay = null;
```

Comment:
Maybe a `TODO` is required here? Or does it not make much sense? In any case we need a clarification.

Reply:
I added `TODO` comments here.